### ISSUE TITLE
problem during training Step4 ?

### ISSUE THREAD

**Issue by DougDong666** [unibas-gravis/Occlusion-Robust-MoFA#16]
> I was also troubled by this problem or bug. Have you solved this problem during training Step4 ?

Yes, I've solved. You should compare the code in the step4 with the code in the step1 and step3, and then fix it.

_Originally posted by @NewtOliver in https://github.com/unibas-gravis/Occlusion-Robust-MoFA/issues/14#issuecomment-1329126412_

**Comment by DougDong666** [comment]
Loss functions in step4 are followed by article. But the loss function in step3 is a classic segmentation loss function. If the loss functions are changed by step3, I don not know why the article needs to mention its own loss fuctions.

**Comment by ChunLLee** [comment]
Hi, step 3 intends to pre-train the segmentation network with the preliminary segmentation masks estimated by our occlusion-robust function, not the ground truth masks.
The losses in step 4 aim to improve the estimation in an unsupervised way.
